### PR TITLE
#46: Get rid off build warnings

### LINKED ISSUE #46: Get rid off build warnings
## Background
There are warnings polluting the build log.

## Feature
Fix the code to remove the warnings.

### PR DESCRIPTION
**Pull request by benedeki**
* fail on compilation warning
* no parallel testing
* Using `JavaConverters` instead of obsolete `JavaConversions`
* Fixed some _ScalaDoc_ links
* Suppressed too chatty logging in tests

Closes #46

### CODE REVIEW

**Review by AdrianOlosutean — DISMISSED** [review]

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code).

I just wish that the common log4j settings in `spark-commons-test/src/test/resources/log4j.properties` and 
`spark-commons/src/test/resources/log4j.properties` could be shared somehow. But I wonder if this can be easily done.

**Comment by lsulak** [comment]
> LGTM (just read the code).
> 
> I just wish that the common log4j settings in `spark-commons-test/src/test/resources/log4j.properties` and `spark-commons/src/test/resources/log4j.properties` could be shared somehow. But I wonder if this can be easily done.

Also, I wonder whether SparkCommonsTestDependenciesReport cannot be shared / implemented only once, since this module appears in the repo twice. I understand the usecase, but I'm not sure if wouldn't be loadable from log4j.properties if this file is in a different module?

**Review by lsulak — APPROVED** [review]

**Comment by benedeki** [comment]
> LGTM (just read the code).
> 
> I just wish that the common log4j settings in `spark-commons-test/src/test/resources/log4j.properties` and `spark-commons/src/test/resources/log4j.properties` could be shared somehow. But I wonder if this can be easily done.

Not sure how to have this file in common. On the other hand the two already differs slightly and chance are they might diverge further with added code.

**Comment by benedeki** [comment]
> > LGTM (just read the code).
> > I just wish that the common log4j settings in `spark-commons-test/src/test/resources/log4j.properties` and `spark-commons/src/test/resources/log4j.properties` could be shared somehow. But I wonder if this can be easily done.
> 
> Also, I wonder whether `SparkCommonsTestDependenciesReport` cannot be shared / implemented only once, since this module appears in the repo twice. I understand the usecase, but I'm not sure if wouldn't be loadable from `log4j.properties` if this file is in a different module?

This was actually solvable by a new trait used in the two pseudo-tests.

File: spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/DependenciesInfo.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  def scalaVersion: String = util.Properties.versionNumberString
+
+  def dependenciesInfo: String = {
+    val line = s"Running Spark $sparkVersion on Scala $scalaVersion"
+    val spaceCount = 1.max(80 - 2 - 1 - line.length) //at least one space, otherwise original length - starting * and space ending * and text length
+    val spaces = " " * spaceCount
+    val result = "\n********************************************************************************\n" +
+                s"* $line$spaces*\n" +
+                 "********************************************************************************"
+    result
```
Comment (dk1844) [inline spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/DependenciesInfo.scala:34]:
```suggestion
  s"""|********************************************************************************
      |* $line$spaces*
      |********************************************************************************"""
    .stripMargin
```

or perhaps even

```suggestion
  s"""|${"*" * 80}
      |* $line$spaces*
      |${"*" * 80}"""
    .stripMargin
```

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code)

File: spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala [hunk 2]
```diff
@@ -92,7 +92,7 @@ object DataFrameImplicits {
     def alignSchema(selector: List[Column]): DataFrame = df.select(selector: _*)
 
     /**
-     * Using utils selector from [[DataFrameSelector.getDataFrameSelector]] aligns the utils of a DataFrame to the selector for operations
+     * Using utils selector from [[StructTypeImplicits.DataFrameSelector.getDataFrameSelector]] aligns the utils of a DataFrame to the selector for operations
```
Comment (Zejnilovic) [inline spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:95]:
```suggestion
     * Using utils selector from [[StructTypeImplicits.DataFrameSelector.getDataFrameSelector]] aligns the utils of 
     * a DataFrame to the selector for operations where utils order might be important (e.g. hashing the whole rows and
     * using except)
```

**Review by Zejnilovic — APPROVED** [review]
sbt compile
and
code reviewed

**Pull request merged**